src/overlays/actors/ovl_Obj_Um/z_obj_um.c

```diff
@@ -46,131 +80,1951 @@ const ActorInit Obj_Um_InitVars = {
+                this->dyna.actor.focus.pos.y - sp30.y,
+                Math3D_XZLength(sp30.x - this->dyna.actor.focus.pos.x, sp30.z - this->dyna.actor.focus.pos.z));
+            this->unk_2FE.x = rot->x + sp3E;
+            this->unk_2FE.y = rot->y;
+            this->unk_2FE.z = rot->z + temp_v0_3;
+            Math_SmoothStepToS(&this->unk_2F8.x, this->unk_2FE.x, 4, 0x3E8, 1);
+            Math_SmoothStepToS(&this->unk_2F8.z, this->unk_2FE.z, 4, 0x3E8, 1);
+            this->unk_2F8.y = rot->y;
+            rot->x = this->unk_2F8.x;
+            rot->z = this->unk_2F8.z;
+        } else {
+            this->unk_2FE.x = rot->x;
+            this->unk_2FE.y = rot->y;
+            this->unk_2FE.z = rot->z;
+            Math_SmoothStepToS(&this->unk_2F8.x, this->unk_2FE.x, 4, 0x3E8, 1);
+            Math_SmoothStepToS(&this->unk_2F8.z, this->unk_2FE.z, 4, 0x3E8, 1);
+            this->unk_2F8.y = rot->y;
+            rot->x = this->unk_2F8.x;
+            rot->z = this->unk_2F8.z;
+        }
+    } else if ((limbIndex == UM_LIMB_WAGON_CART_COVER) && (this->flags & OBJ_UM_FLAG_PLAYING_MINIGAME)) {
+        *dList = NULL;
+    }
+
+    return false;
+}
+
+void ObjUm_SpawnFragments(PlayState* play, Vec3f* potPos) {
+    Vec3f sp8C = { 0.0f, -1.0f, 0.0f };
+    Gfx* potFragments[] = { object_um_DL_000040, object_um_DL_000910, object_um_DL_0011E0 };
+    s32 i;
+    Vec3f sp70;
+
+    EffectSsHitMark_SpawnFixedScale(play, 0, potPos);
+
+    for (i = 0; i < 20; i++) {
+        sp70.x = (Rand_ZeroOne() * 20.0f) - 10.0f;
+        sp70.y = -((Rand_ZeroOne() * 20.0f) - 10.0f);
+        sp70.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
+        EffectSsHahen_Spawn(play, potPos, &sp70, &sp8C, 1, 100, OBJECT_UM, 10,
+                            potFragments[(s32)(Rand_ZeroOne() * ARRAY_COUNT(potFragments))]);
+    }
+}
+
+void ObjUm_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    ObjUm* this = THIS;
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+    Mtx* mtx_s3;
+    Gfx* spFC[] = {
+        NULL, gUmBrokenMinigamePotDL, gUmMinigamePotDL, gUmMinigamePotDL, gUmMinigamePotDL, object_um_DL_0067C0
+    };
+    Gfx* spE4[] = { NULL, NULL, object_um_DL_004B60, object_um_DL_0043E0, NULL, NULL };
+    f32 spCC[] = { 0.0f, 1070.0f, 1070.0f, 1070.0f, 1070.0f, 2100.0f };
+    Vec3f spC0 = gZeroVec3f;
+    Vec3f calcPotPos;
+    f32 spB0;
+
+    spC0.y += 1700.0f;
+
+    if (limbIndex == UM_LIMB_WAGON_CART_BED) {
+        Vec3f spA4 = { 2000.0f, 1070.0f, 0.0f };
+
+        Matrix_MultVec3f(&spA4, &this->unk_308);
+        this->flags |= OBJ_UM_FLAG_DRAWN_FLOOR;
+    }
+
+    if (limbIndex == UM_LIMB_WAGON_CART_BED) {
+        Vec3f sp98 = { 2500.0f, 200.0f, 0.0f };
+
+        Matrix_MultVec3f(&sp98, &this->cartBedPos);
+    }
+
+    if (limbIndex == UM_LIMB_WAGON_CART_BED) {
+        Vec3f* potPos;
+        Vec3f sp88;
+        Vec3s sp80;
+        s32 i;
+        f32 sp70[] = { 2000.0f, 0.0f, -2000.0f };
+
+        if (!i) {}
```

Comment:
```suggestion

        //! FAKE
        if (!i) {}
```
Could it just be `if (i) {}`?